src/common/datasource/src/compression.rs

```diff
@@ -205,3 +215,48 @@ impl From<CompressionType> for FileCompressionType {
         }
     }
 }
+#[cfg(test)]
+mod deserialization_tests {
+    use serde::Deserialize;
+
+    use super::*;
+
+    #[derive(Deserialize)]
+    struct Wrapper {
+        #[serde(default)]
+        compression: CompressionType,
+    }
+
+    #[test]
+    fn test_deserialize_all_variants() {
+        let cases = vec![
+            (r#"{"compression": "gzip"}"#, CompressionType::Gzip),
+            (r#"{"compression": "bzip2"}"#, CompressionType::Bzip2),
+            (r#"{"compression": "xz"}"#, CompressionType::Xz),
+            (r#"{"compression": "zstd"}"#, CompressionType::Zstd),
+            (
+                r#"{"compression": "uncompressed"}"#,
+                CompressionType::Uncompressed,
+            ),
+        ];
+
+        for (json, expected) in cases {
+            let wrapper: Wrapper = serde_json::from_str(json).unwrap();
+            assert_eq!(wrapper.compression, expected, "Failed for json: {}", json);
+        }
+    }
+
+    #[test]
+    fn test_deserialize_empty_string_defaults_to_uncompressed() {
+        let json = r#"{"compression": ""}"#;
```

Comment:
Do we need to support an empty string as compression?

Reply:
Yes, and which is `uncompression`.

Looks like it's used in manifest.

Reply:
The behaviour is the same as `from_str`:
https://github.com/GreptimeTeam/greptimedb/blob/5eb491df126fe34fa55c11e9dabb39e9ed57c955/src/common/datasource/src/compression.rs#L56